src/plugins/mer/meremulatordevice.cpp

```diff
@@ -550,6 +585,8 @@ MerEmulatorDeviceManager::MerEmulatorDeviceManager(QObject *parent)
             this, &MerEmulatorDeviceManager::onDeviceRemoved);
     connect(DeviceManager::instance(), &DeviceManager::deviceListReplaced,
             this, &MerEmulatorDeviceManager::onDeviceListReplaced);
+    connect(MerSettings::instance(), &MerSettings::deviceModelsChanged,
+            this, &MerEmulatorDeviceManager::onDeviceModelsChanged, Qt::QueuedConnection);
```

Comment:
Qt::QueuedConnection needed to ensure that onDeviceModelsChanged executes after onDeviceListReplaced.

Reply:
I was sure an issue like this would arise :\ I do not really like the solution with queued connection - from my experience it causes headaches sooner or later, but I do not have an idea how to fix it *quickly*, so please add a comment here:

```
// Hack: ensure that onDeviceModelsChanged executes after onDeviceListReplaced 
// for the case that user deleted (or edited name of) currently selected device 
// model and immediately pressed Ok in Options.
```

Reply:
Agreed about QueuedConnection. Reworked to different (better) solution.

Reply:
Cool! :)